### DISCUSSION TITLE
Keep the dropdown open on checkbox click

### DISCUSSION THREAD

**Discussion by rawnly** [radix-ui/primitives#1632] [Help] [answered]
Hello! I'd like to keep my dropdown menu open once the user selects an option. Is there any prop that i'm missing? 

I tried the following:
- `stopPropagation` on `CheckBoxItem` click 
- setting `modal` to `false`

## Current behaviour
![CleanShot 2022-08-29 at 12 31 39](https://user-images.githubusercontent.com/16429579/187182504-c890df49-acd1-419b-8e0a-eda7470da62b.gif)

## Expected behaviour
![CleanShot 2022-08-29 at 12 58 56](https://user-images.githubusercontent.com/16429579/187186637-da341e4d-6261-4aac-89e0-cb22e11a2646.gif)

This is accomplished by preventing default `onClick`, but in this way i have to handle the checkbox state without `onCheckedChange` (which is kinda ugly but it does the job).

```tsx
	<Dropdown.Group>
		{options.map(opt => (
			<CheckboxItem
				key={opt}
				checked={data[opt]}
				onClick={e => {
					e.preventDefault();
					setData(s => ({ ...s, [opt]: !s[opt] }));
				}}
				onCheckedChange={checked => setData(s => ({ ...s, [opt]: checked }))}
			>
				{opt}
			</CheckboxItem>
		))}
	</Dropdown.Group>
```

Thanks in advance

**Comment by jjenzz** [comment, marked as answer]
There is [an `onSelect` event](https://www.radix-ui.com/docs/primitives/components/dropdown-menu#item) which you can prevent.

![CleanShot 2022-08-30 at 11 38 22](https://user-images.githubusercontent.com/175330/187416181-96ecc0bd-6635-4956-aeba-f71158e0d315.png)

**Reply by rawnly** [reply]
Thanks I just missed that!
